### ISSUE TITLE
Can't wake Windows 10

### ISSUE THREAD

**Issue by pax0707** [AlexGustafsson/homebridge-wol#32]
### Backstory
I can shut down my win10 PC but wake doesn't work.
I can wake from my router but homebridge doesn't do it. 

### Environment
* os & os version: Win 10 x64
* node version: 9.7.1
* homebridge version: 0.4.38
* homebridge-wol version: 3.2.1

### Log
For logging I removed other plugins from config.

```
{
	"bridge": {
		"name": "HomeBridge",
		"username": "CC:22:3D:E3:CE:20",
		"port": 51877,
		"pin": "031-77-154"
	},
	"accessories": [{
		"accessory": "NetworkDevice",
		"name": "Vobiscum",
		"mac": "<mac-address>",
		"ip": "172.16.1.9",
		"pingInterval": 45,
		"wakeGraceTime": 90,
		"shutdownGraceTime": 60,
		"shutdownCommand": "SHUTDOWN /s /f /t 0 /m 172.16.1.9"
	}]
}
```

```
C:\Users\pax07>homebridge
[2018-3-10 01:43:34] Loaded plugin: homebridge-config-ui-x
[2018-3-10 01:43:34] Registering platform 'homebridge-config-ui-x.config'
[2018-3-10 01:43:34] ---
[2018-3-10 01:43:35] Loaded plugin: homebridge-dacp
[2018-3-10 01:43:35] Registering platform 'homebridge-dacp.DACP'
[2018-3-10 01:43:35] ---
[2018-3-10 01:43:35] Loaded plugin: homebridge-delay-switch
[2018-3-10 01:43:35] Registering accessory 'homebridge-delay-switch.DelaySwitch'
[2018-3-10 01:43:35] ---
[2018-3-10 01:43:35] Loaded plugin: homebridge-plex
[2018-3-10 01:43:35] Registering accessory 'homebridge-plex.Plex'
[2018-3-10 01:43:35] ---
[2018-3-10 01:43:35] Loaded plugin: homebridge-samsung-remote
[2018-3-10 01:43:35] Registering accessory 'homebridge-samsung-remote.SamsungTV'
[2018-3-10 01:43:35] ---
[2018-3-10 01:43:35] Loaded plugin: homebridge-weather-station-extended
[2018-3-10 01:43:35] Registering platform 'homebridge-wunderground-extended.WeatherStation'
[2018-3-10 01:43:36] ---
[2018-3-10 01:43:36] Loaded plugin: homebridge-wol
[2018-3-10 01:43:36] Registering accessory 'homebridge-wol.NetworkDevice'
[2018-3-10 01:43:36] ---
[2018-3-10 01:43:36] Loaded config.json with 1 accessories and 0 platforms.
[2018-3-10 01:43:36] ---
[2018-3-10 01:43:36] Loading 1 accessories...
[2018-3-10 01:43:36] [Vobiscum] Initializing NetworkDevice accessory...
[2018-3-10 01:43:36] [Vobiscum] Stopping pinger
[2018-3-10 01:43:36] [Vobiscum] Starting pinger at an interval of 45000 milli seconds
Setup Payload:
X-HM://0023ITNLUFMCK
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:



























Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ 031-77-154 │
    └────────────┘

  EventedHTTPServer Server listening on port 51877 +0ms
[2018-3-10 01:43:36] Homebridge is running on port 51877.
  EventedHTTPServer [::ffff:172.16.1.30] New connection from client +1s
  EventedHTTPServer [::ffff:172.16.1.30] HTTP server listening on port 49606 +6ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /pair-verify +22ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +3ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 1/2 +5ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +26ms
  EventedHTTPServer [::ffff:172.16.1.13] New connection from client +23ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP server listening on port 49608 +4ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /pair-verify +21ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +7ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 2/2 +3ms
  HAPServer [CC:22:3D:E3:CE:20] Client B68F9B09-30AC-4B70-B01D-378C9CD17925 verification complete +8ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +2ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP request: /pair-verify +1ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +0ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 1/2 +3ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP Response is finished +35ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /accessories +74ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: GET /accessories +5ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +26ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP request: /pair-verify +21ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +5ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 2/2 +4ms
  HAPServer [CC:22:3D:E3:CE:20] Client BE78E569-3CDA-4E67-BC98-782BCB323F0B verification complete +28ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP Response is finished +7ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP request: /accessories +111ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: GET /accessories +3ms
  EventedHTTPServer [::ffff:172.16.1.13] HTTP Response is finished +2ms
  EventedHTTPServer [::ffff:172.16.1.16] New connection from client +1m
  EventedHTTPServer [::ffff:172.16.1.16] HTTP server listening on port 50058 +3ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP request: /pair-verify +5ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +1ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 1/2 +1ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP Response is finished +5ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP request: /pair-verify +17ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: POST /pair-verify +0ms
  HAPServer [CC:22:3D:E3:CE:20] Pair verify step 2/2 +1ms
  HAPServer [CC:22:3D:E3:CE:20] Client 5FB36FAD-D135-4C33-BB69-CEE446881E18 verification complete +2ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP Response is finished +10ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP request: /accessories +6ms
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: GET /accessories +3ms
  EventedHTTPServer [::ffff:172.16.1.16] HTTP Response is finished +15ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /characteristics?id=12.10 +5s
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: GET /characteristics?id=12.10 +3ms
  Accessory [HomeBridge] Getting value for Characteristic "On" +4ms
  Accessory [HomeBridge] Got Characteristic "On" value: false +1ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +2ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /characteristics?id=12.10 +7s
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: GET /characteristics?id=12.10 +3ms
  Accessory [HomeBridge] Getting value for Characteristic "On" +3ms
  Accessory [HomeBridge] Got Characteristic "On" value: false +4ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +8ms
  EventedHTTPServer [::ffff:172.16.1.30] HTTP request: /characteristics +5s
  HAPServer [CC:22:3D:E3:CE:20] HAP Request: PUT /characteristics +11ms
  Accessory [HomeBridge] Processing characteristic set: [{"aid":12,"iid":10,"value":true}] +4ms
  Accessory [HomeBridge] Setting Characteristic "On" to value true +2ms
[2018-3-10 01:45:23] [Vobiscum] NetworkDevice awake cycle started for "Vobiscum" (172.16.1.9)
[2018-3-10 01:45:23] [Vobiscum] Stopping pinger
[2018-3-10 01:45:23] [Vobiscum] Starting pinger again in 90000 milli seconds
[2018-3-10 01:45:23] [Vobiscum] Attempting to wake up "Vobiscum" (172.16.1.9)
[2018-3-10 01:45:23] [Vobiscum] NetworkDevice "Vobiscum" (172.16.1.9) went from status "Offline" to "Waking Up"
  EventedHTTPServer [::ffff:172.16.1.30] HTTP Response is finished +22ms
[2018-3-10 01:46:53] [Vobiscum] Stopping pinger
[2018-3-10 01:46:53] [Vobiscum] Starting pinger at an interval of 45000 milli seconds
[2018-3-10 01:46:54] [Vobiscum] Pinger toggled state change
[2018-3-10 01:46:54] [Vobiscum] NetworkDevice "Vobiscum" (172.16.1.9) went from status "Waking Up" to "Offline"
[2018-3-10 01:52:28] Got SIGINT, shutting down Homebridge...
Terminate batch job (Y/N)? y
```

**Comment by AlexGustafsson** [comment]
There seems to be no configuration error and I can't see any mistake in the logs.

Sometimes it's the simple things. Can you verify that the computer running homebridge-wol and the computer that you're trying to wake are on the same network? That is, with the same network address and subnet mask (192.168.1.x 255.255.255.0). Also, please verify that the MAC-adress you're trying to wake is correct.

**Comment by pax0707** [comment]
My network is 172.16.1.0 255.255.255.0.

The router can wake both PCs, MAC is correct.

When I check with Wake-on-LAN Packet Sniffer I get it from the router but not from homebridge.

**Comment by AlexGustafsson** [comment]
Does any other Wake-on-LAN utility work from the computer running homebridge? Can you for example run `etherwake MAC` or `wakeonlan MAC` (given that the packages are available)?

**Comment by pax0707** [comment]
```
Packet recieved from 172.16.1.7 on port 58781
at 01:20:45 packet length : 102
00000000 : FF FF FF FF FF FF 00 1F D0 A2 A6 84 00 1F D0 A2 
00000010 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000020 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000030 : 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 
00000040 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000050 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000060 : 00 1F D0 A2 A6 84 
Packet recieved from 172.16.1.7 on port 58782
at 01:20:47 packet length : 102
00000000 : FF FF FF FF FF FF 00 1F D0 A2 A6 84 00 1F D0 A2 
00000010 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000020 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000030 : 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 
00000040 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000050 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000060 : 00 1F D0 A2 A6 84 
Packet recieved from 172.16.1.1 on port 32769
at 01:25:17 packet length : 102
00000000 : FF FF FF FF FF FF 00 1F D0 A2 A6 84 00 1F D0 A2 
00000010 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000020 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000030 : 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 
00000040 : A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F 
00000050 : D0 A2 A6 84 00 1F D0 A2 A6 84 00 1F D0 A2 A6 84 
00000060 : 00 1F D0 A2 A6 84

**Comment by AlexGustafsson** [comment]
@pax0707 I think I've found the problem. The dependency in charge of sending the magic packet to wake up a device sends out a broadcast UDP packet on the network and assumes that the netmask is `255.255.255.255`. This is an issue in your case since the broadcast will never be received by any host on the network.

Could you please validate this by making the following change?

[Line 139 in lib/network-device.js](https://github.com/AlexGustafsson/homebridge-wol/blob/master/lib/network-device.js#L139):
```
// From:
wol.wake(this.config.mac, error => {
// To:
wol.wake(this.config.mac, {address: '255.255.255.0'}, error => {
```

You should then be able to start homebridge and see if it works.

**Comment by pax0707** [comment]
```
[2018-3-24 11:40:02] [Vobiscum] NetworkDevice awake cycle started for "Vobiscum" (172.16.1.9)
[2018-3-24 11:40:02] [Vobiscum] Attempting to wake up "Vobiscum" (172.16.1.9)
[2018-3-24 11:40:02] [Vobiscum] NetworkDevice "Vobiscum" (172.16.1.9) went from status "Offline" to "Waking Up"
[2018-3-24 11:40:02] [Vobiscum] An error occured while waking "Vobiscum" (172.16.1.9, 00:1F:D0:A2:A6:84): Error: send ENETUNREACH 255.255.255.0:9
[2018-3-24 11:40:02] [Vobiscum] NetworkDevice "Vobiscum" (172.16.1.9) went from status "Waking Up" to "Offline"
```

**Comment by AlexGustafsson** [comment]
@pax0707 Did you ever try the steps I mentioned in one of the comments above ([this one](https://github.com/AlexGustafsson/homebridge-wol/issues/32#issuecomment-375398863))?

Those steps will help you test if the module for wake on lan.

**Comment by kloshar4o** [comment]
For some reason the currentState variable doesnt change to false, and the if statement doesnt execute on [line 104 in lib/network-device.js](https://github.com/AlexGustafsson/homebridge-wol/blob/55f587e51d8d1c2711f0f484490fc8b6e984f34b/lib/network-device.js#L104) 

I dont see how to debug it, so i just changed status when the PC is shutting down from **ShuttingDown** to **Offline** on line 158

**Comment by AlexGustafsson** [comment]
@kloshar4o I've mentioned you in the issue #35. Could you please head over there and follow the instructions to test a potential fix?

**Comment by pax0707** [comment]
Sorry for long time to reply. 
Comparing what my router was sending to what plugin was sending with packet sniffer address must be 172.16.1.255. 
Kinda logical, since that's the broadcast address for that range.

**Comment by AlexGustafsson** [comment]
I added an option to specify broadcast address (`broadcastAddress`). The option is available in 3.2.2 now published to NPM.

Thanks for helping me find and solve the issue!
